virtme/commands/run.py

```
@@ -1282,25 +1282,25 @@ def do_it() -> int:
     if args.graphics is None and not args.script_sh and not args.script_exec:
         qemuargs.extend(["-echr", "1"])
 
-        if args.verbose:
-            # Check if we have permission to access the current stderr.
-            if not can_access_file("/proc/self/fd/2"):
-                print(
-                    "ERROR: not a valid pts, try to run vng inside tmux or screen",
-                    file=sys.stderr,
-                )
-                sys.exit(1)
+        # Check if we have permission to access the current stderr.
+        if not can_access_file("/proc/self/fd/2"):
```

Do you know when this error could happen?

I mean: if it can happen in "normal" situations, maybe no need to exit except in verbose mode? (so the previous `if args.verbose` could be replaced by `if can_access_file("/proc/self/fd/2")` followed by a `elif args.verbose: print(); exit();`

---

> Do you know when this error could happen?

Oh that's a good point, yes, it can happen when running vng in certain containers.

> 
> I mean: if it can happen in "normal" situations, maybe no need to exit except in verbose mode? (so the previous `if args.verbose` could be replaced by `if can_access_file("/proc/self/fd/2")` followed by a `elif args.verbose: print(); exit();`

Hm... but in this way we would implicitly suppress the errors when stderr isn't available. Maybe we can do that and also print a warning to stdout that we're implicitly suppressing errors because stderr isn't accessible?

---

> > Do you know when this error could happen?
> 
> Oh that's a good point, yes, it can happen when running vng in certain containers.

OK, then probably best not to break this.

> > I mean: if it can happen in "normal" situations, maybe no need to exit except in verbose mode? (so the previous `if args.verbose` could be replaced by `if can_access_file("/proc/self/fd/2")` followed by a `elif args.verbose: print(); exit();`
> 
> Hm... but in this way we would implicitly suppress the errors when stderr isn't available. Maybe we can do that and also print a warning to stdout that we're implicitly suppressing errors because stderr isn't accessible?

Good point!
Or maybe in this case, it is fine to print the panic messages to stdout? These messages should be exceptional, no?

---

> > Hm... but in this way we would implicitly suppress the errors when stderr isn't available. Maybe we can do that and also print a warning to stdout that we're implicitly suppressing errors because stderr isn't accessible?
> 
> Good point! Or maybe in this case, it is fine to print the panic messages to stdout? These messages should be exceptional, no?

Alright, I've updated the PR, now we're never exiting, if we don't have access to stderr via procfs (e.g., when we run vng inside a container with restricted permissions), we print a warning, instead of an error, and suppress the kernel messages redirection. Otherwise, if we have access to stderr via procfs, we redirect only the panic/oops trace to stderr by default and with verbose we redirect all the kernel log messages to stderrr.

I think in this way we won't break anything, we can get the panic/oops when we run interactively and with `--verbose` everything remains the same.